### PR TITLE
Add a patch URL to change nickname

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
create an endpoint that when hit changes the user's server nickname as per the RDS backend.

endpoint Method: `PATCH`
Headers: `JWT` signed with `BOT_PUBLIC_KEY` stored in the RDS backend
body: 
```json
{
    id: <Discord id of the user>,
    username: <RDS Username>
}
```
endpoint URL: `Cloudflare handler link`
path: `/name/update`

When the endpoint is hit with the above-mentioned attributes the server nickname of the user will be changed to the username passed in the request body.

The authorization header includes jwt which is verified to check if the request is from valid source.

Edge Cases
- User may change their server names after the patch request
    - `solution`: stop allowing users to change nicknames in the server. Make sure only bot is able to do that.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -20,6 +21,8 @@ router.get("/", async () => {
   });
 });
 
+router.patch("/change-nickname", changeNickname);
```
Comment (DashDeipayan) [inline src/index.ts:24]: Can we have a better naming of the routes. Routes directly named with functionality is not recommended. Something like `\name\<something>` fits well but feel free to come up with your own.

File: src/utils/updateNickname.ts [hunk 2]
```diff
@@ -19,10 +19,7 @@ export async function updateNickName(
...
     });
     if (nameChangeResponse.ok) {
-      return await nameChangeResponse.json();
-    } else {
-      console.log(nameChangeResponse.json());
-      return INTERNAL_SERVER_ERROR;
+      return NAME_CHANGED;
     }
   } catch (error) {
     console.log(error);
```
Comment (DashDeipayan) [inline src/utils/updateNickname.ts:25]: Console messages shouldn't be present in prod code

File: tests/unit/utils/updateNickname.test.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  const mockEnv = {
+    BOT_PUBLIC_KEY: "xyz",
+    DISCORD_GUILD_ID: "123",
+    DISCORD_TOKEN: "abc",
+  };
+
+  const mockData = { discordId: "12345678910111213", nickname: "jhon" };
+
+  test("updatenickname fetch is called with expected parameters", async () => {
+    const data = { nick: mockData.nickname };
```
Comment (DashDeipayan) [inline tests/unit/utils/updateNickname.test.ts:19]: `nick` doesn't sound very convincing haha. We can name something better i hope 😂

File: src/constants/responses.ts [hunk 4]
```diff
@@ -23,3 +23,5 @@ export const RETRY_COMMAND =
   "Oops, we didn't catch that! Please use the command again.";
 
 export const ROLE_ADDED = "Role added successfully";
+
+export const NAME_CHANGED = "User nickname change successful";
```
Comment (DashDeipayan) [inline src/constants/responses.ts:27]: The statement `User nickname changed successfully` sounds better as a response and we have also followed similar responses previously. That is better I feel, isn't it?

**Review by DashDeipayan — CHANGES_REQUESTED** [review]

File: src/utils/updateNickname.ts (hunk 2 above)
Comment (RitikJaiswal75) [reply src/utils/updateNickname.ts:25]: as you can see it is a deleted line

File: tests/unit/utils/updateNickname.test.ts (hunk 3 above)
Comment (RitikJaiswal75) [reply tests/unit/utils/updateNickname.test.ts:19]: That cant be chosen by the developer discord has a pre defined key that needs to be send as body.
https://discord.com/developers/docs/resources/guild#modify-guild-member

File: src/constants/responses.ts (hunk 4 above)
Comment (RitikJaiswal75) [reply src/constants/responses.ts:27]: I am not sure, both convey the same message.

Comment (DashDeipayan) [reply src/constants/responses.ts:27]: Yes, but one looks like an action response and the other seems like a status update. Just a small thing

File: tests/unit/utils/updateNickname.test.ts (hunk 3 above)
Comment (DashDeipayan) [reply tests/unit/utils/updateNickname.test.ts:19]: oh cool, that's hilarious.

File: src/utils/updateNickname.ts (hunk 2 above)
Comment (RitikJaiswal75) [reply src/utils/updateNickname.ts:25]: My bad removing it

File: src/index.ts (hunk 1 above)
Comment (RitikJaiswal75) [reply src/index.ts:24]: I am trying to keep it in a single name to avoid multiple nesting

File: src/utils/updateNickname.ts (hunk 2 above)
Comment (RitikJaiswal75) [reply src/utils/updateNickname.ts]: done

File: src/index.ts (hunk 1 above)
Comment (DashDeipayan) [reply src/index.ts:24]: Everything related to name and namechanges must go under `name` or `<something-else>` path. They will be differentiated based on HTTP methods, sub-paths and path-params

- https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/

**Review by DashDeipayan — APPROVED** [review]
LGTM

**Pull request merged**